src/ServiceBus/ServiceBus/Cmdlets/NetworkRuleSet/AddAzureServiceBusVirtualNetworkrule.cs

```diff
@@ -0,0 +1,85 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.ServiceBus.Models;
+using System.Management.Automation;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using System.Collections.Generic;
+
+namespace Microsoft.Azure.Commands.ServiceBus.Commands.NetworkruleSet
+{
+    /// <summary>
+    /// 'New-AzureRmEventHubIpfilterRule' Cmdlet creates a new IPFilterRule
+    /// </summary>
+    [Cmdlet("Add", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "ServiceBusVirtualNetworkRule", DefaultParameterSetName = VirtualNetworkRulePropertiesParameterSet, SupportsShouldProcess = true), OutputType(typeof(PSNetworkRuleSetAttributes))]
+    public class NewAzureEventHubVNetRule : AzureServiceBusCmdletBase
+    {
+        [Parameter(Mandatory = true, ParameterSetName = VirtualNetworkRulePropertiesParameterSet, ValueFromPipelineByPropertyName = true, Position = 0, HelpMessage = "Resource Group Name")]
+        [Parameter(Mandatory = true, ParameterSetName = VirtualNetworkRuleInputObjectParameterSet, ValueFromPipelineByPropertyName = true, Position = 0, HelpMessage = "Resource Group Name")]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+         public string ResourceGroupName { get; set; }
+
+        [Parameter(Mandatory = true, ParameterSetName = VirtualNetworkRulePropertiesParameterSet, ValueFromPipelineByPropertyName = true, Position = 1, HelpMessage = "Namespace Name")]
+        [Parameter(Mandatory = true, ParameterSetName = VirtualNetworkRuleInputObjectParameterSet, ValueFromPipelineByPropertyName = true, Position = 1, HelpMessage = "Namespace Name")]
+        [ValidateNotNullOrEmpty]
+        [Alias(AliasNamespaceName)]
+        public string Name { get; set; }        
+
+        [Parameter(Mandatory = true, ParameterSetName = VirtualNetworkRulePropertiesParameterSet, ValueFromPipelineByPropertyName = true, Position = 2, HelpMessage = "Subnet Resource ID")]
+        [ValidateNotNullOrEmpty]
+        public string SubnetId { get; set; }
+
+        [Parameter(Mandatory = false, ParameterSetName = VirtualNetworkRulePropertiesParameterSet, ValueFromPipelineByPropertyName = true, HelpMessage = "ARM ID of Subnet")]
```

Comment:
@v-Ajnava _minor_: you can remove `ValueFrompipelineByPropertyName` from all `SwitchParameter` parameters

This comment applies to all `SwitchParameter` parameters in this PR that have this property.